x-pack/plugins/apm/public/components/shared/transaction_action_menu/transaction_action_menu.test.tsx

```diff
@@ -375,8 +375,10 @@ describe('TransactionActionMenu component', () => {
       const getFilterKeyValue = (key: string) => {
         return {
           [(component.getAllByText(key)[0] as HTMLOptionElement).text]: (
-            component.getAllByTestId(`${key}.value`)[0] as HTMLInputElement
-          ).value,
+            component
+              .getByTestId(`${key}.value`)
+              .querySelector('span') as HTMLSpanElement
+          ).textContent,
```

Comment:
Thanks for improving this selector. More understandable now!
Still, perhaps we should add a `test-data-subject` attribute to make the selector more clear. At a glance I can't see what this is targeting.

Reply:
https://github.com/elastic/kibana/pull/122648/commits/5b5facf2ba46d594ae32db4c1bd116a44520bdb4